### DISCUSSION TITLE
With-knex example. Error on yarn migrate:latest

### DISCUSSION THREAD

**Discussion by sakhmedbayev** [vercel/next.js#19706] [Help]
@ashconnell, I am trying to run `yarn migrate:latest` command after initializing my project using "with-knex" example, getting the following error

```
error: no pg_hba.conf entry for host "92.22.111.10", user "sdsds", database "sdsd", SSL off
    at Connection.parseE (/Users/me/np/mine/node_modules/pg/lib/connection.js:553:11)
    at Connection.parseMessage (/Users/me/np/mine/node_modules/pg/lib/connection.js:378:19)
    at Socket.<anonymous> (/Users/me/np/mine/telosmedia-platform/node_modules/pg/lib/connection.js:119:22)
    at Socket.emit (events.js:314:20)
    at addChunk (_stream_readable.js:298:12)
    at readableAddChunk (_stream_readable.js:273:9)
    at Socket.Readable.push (_stream_readable.js:214:10)
    at TCP.onStreamRead (internal/stream_base_commons.js:188:23)
```

node version v12.19.0
postgres (PostgreSQL) 13.1
postgresql db instance is hosted on the heroku free app

**Comment by ashconnell** [comment]
Hi @sakhmedbayev, 

The `with-knex` example is there to demonstrate how to get knex and a local postgres database to **work with Next.js**

I'm not a support person for how to connect knex to your arbitrary cloud hosted database. That has nothing to do with Next.js.

That being said, the error you're seeing means your database is not authorizing the connection. This could be because the IP isn't whitelisted, the credentials are incorrect or it's only accepting SSL connections.

Feel free to post your solution here in case it helps someone else in the future

**Reply by sakhmedbayev** [reply]
I understand, but you need to realize that the example does not work in a production environment as there is no way to run a local postgres database in vercel. 

I managed to make it work, however, with several changes:

1. set `pg` to the version of "7.4.3",
2. set a connection object rather than a connection string:

```
export const dbConnectionConfig = {
  host: DB_HOST,
  user: DB_USER,
  password: DB_PASSWORD,
  database: DB_DATABASE,
  ssl: DB_SSL, // true
  multipleStatements: true, // not sure if this is necessary
  charset: "utf8",
};
```

**Reply by ashconnell** [reply]
The examples for Next.js show you how to get started using certain libraries. How to integrate them with Next.js. That's all they are for. I put this here as a working reference. A local database is the easiest way to demonstrate this. Asking people to pay for a cloud based managed postgres to demonstrate something is ludicrous.

> you need to realize that the example does not work in a production environment

I've actually been using this in a production environment for 4+ months. It just so happens to work with Digital Ocean's managed postgres without any modifications.

**Reply by sakhmedbayev** [reply]
Thanks, @ashconnell! I did not mean to blame you.
